connector/src/main/java/org/geysermc/connector/utils/LoginEncryptionUtils.java

```diff
@@ -132,7 +132,10 @@ private static void encryptConnectionWithCert(GeyserConnector connector, GeyserS
             JWSObject clientJwt = JWSObject.parse(clientData);
             EncryptionUtils.verifyJwt(clientJwt, identityPublicKey);
 
-            session.setClientData(JSON_MAPPER.convertValue(JSON_MAPPER.readTree(clientJwt.getPayload().toBytes()), BedrockClientData.class));
+            JsonNode clientDataJson = JSON_MAPPER.readTree(clientJwt.getPayload().toBytes());
+            BedrockClientData data = JSON_MAPPER.convertValue(clientDataJson, BedrockClientData.class);
+            data.setJsonData(clientDataJson);
```

Comment:
So this I'm not a fan of - we're storing the JSON client data *and* deserialized fields for each. Can we find a way to only store one?